fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2025 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.server.coordinator;
+
+import com.alibaba.fluss.annotation.VisibleForTesting;
+import com.alibaba.fluss.exception.FlussRuntimeException;
+import com.alibaba.fluss.metadata.TableInfo;
+import com.alibaba.fluss.metadata.TablePath;
+import com.alibaba.fluss.server.utils.timer.DefaultTimer;
+import com.alibaba.fluss.server.utils.timer.Timer;
+import com.alibaba.fluss.server.utils.timer.TimerTask;
+import com.alibaba.fluss.utils.concurrent.ShutdownableThread;
+import com.alibaba.fluss.utils.types.Tuple2;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static com.alibaba.fluss.utils.concurrent.LockUtils.inLock;
+
+/**
+ * A manager to manage the tables to be tiered.
+ *
+ * <p>For a lake table to be tiered, when created, it wil be put into this manager and scheduled to
+ * be tiered by tiering services.
+ *
+ * <p>The state machine of table to be tiered is as follows:
+ *
+ * <pre>{@code
+ * New -> Scheduled
+ * Scheduled --> |after lake tiering interval| Pending
+ * Pending --> |lake tiering service request table| Tiering
+ * Tiering --> |once lake tiering finish| Scheduled
+ * Tiering --> |lake tiering heartbeat timout| Scheduled
```

Comment:
Add description for each status. Besides, I didn't see any state machine for the tables. I think that's important to judge what's the situation of the current table.

Reply:
Now, I add the state machine to track state transmition and current state.